pkg/drivers/jetstream/jetstream.go

```diff
@@ -90,6 +128,92 @@ func New(ctx context.Context, connection string) (server.Backend, error) {
 	}, nil
 }
 
+// parseNatsConnection returns nats connection url, bucketName and []nats.Option, error
+func parseNatsConnection(dsn string, tlsInfo tls.Config) (string, string, []nats.Option, error) {
+
+	o := make([]nats.Option, 0)
+
+	bucketName := kineBucket
+
+	connections := strings.Split(dsn, ",")
+
+	connBuilder := strings.Builder{}
+	for idx, c := range connections {
+		if idx > 0 {
+			connBuilder.WriteString(",")
+		}
+		u, err := url.Parse(c)
+		if err != nil {
+			return "", "", nil, err
+		}
+		if u.Scheme != "nats" {
+			return "", "", nil, fmt.Errorf("invalid connection string=%s", c)
+		}
+		connBuilder.WriteString("nats://")
+		if u.User != nil && idx == 0 {
+			userInfo := strings.Split(u.User.String(), ":")
+			if len(userInfo) > 1 {
+				o = append(o, nats.UserInfo(userInfo[0], userInfo[1]))
+			} else {
+				o = append(o, nats.Token(userInfo[0]))
+			}
+		}
+		connBuilder.WriteString(u.Host)
+	}
+
+	u, err := url.Parse(connections[0])
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	queryMap, err := url.ParseQuery(u.RawQuery)
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	if b, ok := queryMap["bucket"]; ok {
+		bucketName = b[0]
+	}
+
+	clientCertFile := ""
+	clientKeyFile := ""
+
+	if tlsInfo.CertFile != "" {
+		clientCertFile = tlsInfo.CertFile
+	}
+
+	if tlsInfo.KeyFile != "" {
+		clientKeyFile = tlsInfo.KeyFile
+	}
+
+	if clientCertFile != "" && clientKeyFile != "" {
+		o = append(o, nats.ClientCert(clientCertFile, clientKeyFile))
+	}
+
+	if tlsInfo.CAFile != "" {
+		o = append(o, nats.RootCAs(tlsInfo.CAFile))
+	}
+
+	if ctxOpt, ok := queryMap["context"]; ok {
+		logrus.Infof("loading nats context=%s", ctxOpt[0])
+		natsContext, err := natscontext.New(ctxOpt[0], true)
+		if err != nil {
+			return "", "", nil, err
+		}
+		// command line options provided to kine will override the file
+		// https://github.com/nats-io/jsm.go/blob/v0.0.29/natscontext/context.go#L257
+		// allows for user, creds, nke, token, certifcate, ca, inboxprefix from the context.json
+		natsClientOpts, err := natsContext.NATSOptions(o...)
```

Comment:
NATS Context also defines [ServerURL](https://github.com/nats-io/jsm.go/blob/48ea938560dce28b1170ed7d909489454de67792/natscontext/context.go#L451), but that seems to be ignored here...

I think that if a context is used, then the URLs should _only_ come from the context - throw an error if `host` or `port` is set

So a context url would be `nats://?context=/etc/nats.json`

Reply:
Seems reasonable to me

Reply:
Updated to force this scheme